pkg/provider/provider.go

```diff
@@ -1,221 +1,393 @@
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_CLIENT_IP", nil),
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Description: "Either http or https, defaults to https. Can also be sourced from the `SNOWFLAKE_PROTOCOL` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PROTOCOL", nil),
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					switch strings.ToLower(val.(string)) {
+					case "http", "https":
+						return nil, nil
+					default:
+						errs := append(errs, fmt.Errorf("%q must be one of http or https", key))
+						return warns, errs
+					}
+				},
+			},
+			"host": {
+				Type:        schema.TypeString,
+				Description: "Supports passing in a custom host value to the snowflake go driver for use with privatelink. Can also be sourced from the `SNOWFLAKE_HOST` environment variable. ",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_HOST", nil),
+			},
+			"port": {
+				Type:        schema.TypeInt,
+				Description: "Support custom port values to snowflake go driver for use with privatelink. Can also be sourced from the `SNOWFLAKE_PORT` environment variable. ",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PORT", nil),
+			},
+			"authenticator": {
+				Type:        schema.TypeString,
+				Description: "Specifies the [authentication type](https://pkg.go.dev/github.com/snowflakedb/gosnowflake#AuthType) to use when connecting to Snowflake. Valid values include: Snowflake, OAuth, ExternalBrowser, Okta, JWT, TokenAccessor, UsernamePasswordMFA. Can also be sourced from the `SNOWFLAKE_AUTHENTICATOR` environment variable.",
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_AUTHENTICATOR", nil),
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					switch val.(string) {
+					case "Snowflake", "OAuth", "ExternalBrowser", "Okta", "JWT", "TokenAccessor", "UsernamePasswordMFA":
+						return nil, nil
+					default:
+						errs := append(errs, fmt.Errorf("%q must be one of Snowflake, OAuth, ExternalBrowser, Okta, JWT, TokenAccessor or UsernamePasswordMFA", key))
+						return warns, errs
+					}
+				},
+			},
+			"passcode": {
+				Type:          schema.TypeString,
+				Description:   "Specifies the passcode provided by Duo when using multi-factor authentication (MFA) for login. Can also be sourced from the `SNOWFLAKE_PASSCODE` environment variable. ",
+				Optional:      true,
+				ConflictsWith: []string{"passcode_in_password"},
+				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PASSCODE", nil),
+			},
+			"passcode_in_password": {
+				Type:          schema.TypeBool,
+				Description:   "False by default. Set to true if the MFA passcode is embedded in the login password. Appends the MFA passcode to the end of the password. Can also be sourced from the `SNOWFLAKE_PASSCODE_IN_PASSWORD` environment variable. ",
```

Comment:
description states that this is optional by default, but the default is taken from env variable

Reply:
only if it isn't nil. And the default value is from the config, not from us setting the default intentionally. I just searched through the gosnowflake package to find out what the defaults are.